lightning/src/ln/channelmanager.rs

```diff
@@ -3931,23 +3926,19 @@ where
 		if let Err((err_msg, err_code)) = check_incoming_htlc_cltv(
 			cur_height, next_packet_details.outgoing_cltv_value, msg.cltv_expiry
 		) {
-			let chan_update_opt = self.do_funded_channel_callback(next_packet_details.outgoing_scid, |chan: &mut Channel<SP>| {
-				self.get_channel_update_for_onion(next_packet_details.outgoing_scid, chan).ok()
-			}).flatten();
-			return Err((err_msg, err_code, chan_update_opt));
+			return Err((err_msg, err_code));
 		}
 
 		Ok(())
 	}
 
 	fn htlc_failure_from_update_add_err(
 		&self, msg: &msgs::UpdateAddHTLC, counterparty_node_id: &PublicKey, err_msg: &'static str,
-		mut err_code: u16, chan_update: Option<msgs::ChannelUpdate>, is_intro_node_blinded_forward: bool,
+		err_code: u16, is_intro_node_blinded_forward: bool,
 		shared_secret: &[u8; 32]
 	) -> HTLCFailureMsg {
-		let mut res = VecWriter(Vec::with_capacity(chan_update.serialized_length() + 2 + 8 + 2));
-		if chan_update.is_some() && err_code & 0x1000 == 0x1000 {
-			let chan_update = chan_update.unwrap();
+		let mut res = VecWriter(Vec::with_capacity(8 + 2));
```

Comment:
```suggestion
		// at capacity, we write fields `htlc_msat` and `len`
		let mut res = VecWriter(Vec::with_capacity(8 + 2));
```
Is this enough ?

Reply:
Yeah probably is enough! but I am not sure this is clear enough? because the message is still `channel_update_size` + `htlc_msat` + `len`.

>Nodes which do not provide a
`channel_update` are expected to set the `channel_update` `len` field to zero.

And not sure if the serializing code will expect some `0` value for the size of the array?

Reply:
I believe according to the spec, the message is just `htlc_msat` + `len`, where `len` is set to `0`. So 8 bytes + 2 bytes in total.